src/libOpenImageIO/imagebufalgo_muldiv.cpp

```diff
@@ -91,6 +149,8 @@ ImageBufAlgo::mul(ImageBuf& dst, Image_or_Const A_, Image_or_Const B_, ROI roi,
     pvt::LoggedTimer logtime("IBA::mul");
     if (A_.is_img() && B_.is_img()) {
         const ImageBuf &A(A_.img()), &B(B_.img());
+        if (A.nchannels() == 1 || B.nchannels() == 1)
+            return scale(dst, A, B, roi, nthreads);
```

Comment:
This is a behaviour changes for mul(). Maybe that's fine, but we should certainly document it, and maybe think hard about if we want it.

Are we prepared to say that any time somebody multiplies an n-channel image by a 1-channel image, they mean to scale all channels by that one value? Is there ever a case where people would not want that? (I honestly don't know.)

Reply:
I does look weird when the alpha is scaled, as you mentioned, but on the other hand this behaviour is consistent with mul() used with a single scalar.

Reply:
I think that's fine.

I wonder... if maybe we should get in the habit of adding a KWArgs parameter to all new IBA functions (and, slowly, adding them to exiting ones when the occasion pops up that we have to modify them for other reasons), even if we don't have any keyword args to respond to at this moment, as a placeholder for future expansion without needing to alter the ABIs or add more functions. In this case, that would make it more palatable to pick a reasonable default behavior (just scale all channels) for now and then in the future, we can very easily accommodate an option that would cause the alpha to not be scaled.

Reply:
Let's not add it to mul (which would either break compatibility, or cause yet another set of mul versions to be added) until we have a desired use.

But let's add it to scale() now -- it costs nothing and has no compatibility issues for a "new" IBA function, and future-proofs the API.

Reply:
I've added a KWArgs parameter to scale(), and removed the changes from mix() for now.